drivers/gles3/rasterizer_canvas_gles3.cpp

```diff
@@ -1236,13 +1271,41 @@ void RasterizerCanvasGLES3::_render_batch(Light *p_lights, uint32_t p_index) {
 					break;
 				}
 
-				multimesh_buffer = mesh_storage->multimesh_get_gl_buffer(multimesh);
-				multimesh_stride = mesh_storage->multimesh_get_stride(multimesh);
-				multimesh_color_offset = mesh_storage->multimesh_get_color_offset(multimesh);
-				multimesh_uses_color = mesh_storage->multimesh_uses_colors(multimesh);
-				multimesh_uses_custom_data = mesh_storage->multimesh_uses_custom_data(multimesh);
+				instance_buffer = mesh_storage->multimesh_get_gl_buffer(multimesh);
+				instance_stride = mesh_storage->multimesh_get_stride(multimesh);
+				instance_color_offset = mesh_storage->multimesh_get_color_offset(multimesh);
+				instance_uses_color = mesh_storage->multimesh_uses_colors(multimesh);
+				instance_uses_custom_data = mesh_storage->multimesh_uses_custom_data(multimesh);
+
 			} else if (state.canvas_instance_batches[p_index].command_type == Item::Command::TYPE_PARTICLES) {
-				// Do nothing for now.
+				const Item::CommandParticles *pt = static_cast<const Item::CommandParticles *>(state.canvas_instance_batches[p_index].command);
+				RID particles = pt->particles;
+				mesh = particles_storage->particles_get_draw_pass_mesh(particles, 0);
+
+				ERR_BREAK(particles_storage->particles_get_mode(particles) != RS::PARTICLES_MODE_2D);
+				particles_storage->particles_request_process(particles);
+
+				if (particles_storage->particles_is_inactive(particles)) {
+					break;
+				}
+
+				RenderingServerDefault::redraw_request(); // Active particles means redraw request.
+
+				int dpc = particles_storage->particles_get_draw_passes(particles);
+				if (dpc == 0) {
+					break; // Nothing to draw.
+				}
+
+				instance_count = particles_storage->particles_get_amount(particles);
+				instance_buffer = particles_storage->particles_get_gl_buffer(particles);
+				instance_stride = 12;
```

Comment:
I'm wondering if it would make sense for prosperity, to add some info about the buffer layout here so we know why the stride is hardcoded to 12 and offset to 8. Also I wonder if these should be constants.

If every someone changes the buffer, they may not realise these values change.

Reply:
Updated!